publisher/publisher.go

```diff
@@ -6,273 +6,127 @@
 package publisher
 
 import (
-	"bytes"
-	"crypto/ecdsa"
-	"crypto/sha256"
 	"crypto/x509"
 	"encoding/base64"
-	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"net/http"
-	"strconv"
 	"strings"
-	"time"
+
+	ct "github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/google/certificate-transparency/go"
+	ctClient "github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/google/certificate-transparency/go/client"
 
 	"github.com/letsencrypt/boulder/core"
 	blog "github.com/letsencrypt/boulder/log"
-	"github.com/letsencrypt/boulder/sa"
 )
 
-// LogDescription tells you how to connect to a log and verify its statements.
-type LogDescription struct {
-	ID        string
-	URI       string
-	PublicKey *ecdsa.PublicKey
+// Log contains the CT client and signature verifier for a particular CT log
+type Log struct {
+	Client   *ctClient.LogClient
+	Verifier *ct.SignatureVerifier
 }
 
-type rawLogDescription struct {
+// LogDescription something something
+type LogDescription struct {
 	URI       string `json:"uri"`
 	PublicKey string `json:"key"`
 }
 
-// UnmarshalJSON parses a simple JSON format for log descriptions.  Both the
-// URI and the public key are expected to be strings.  The public key is a
-// base64-encoded PKIX public key structure.
-func (logDesc *LogDescription) UnmarshalJSON(data []byte) error {
-	var rawLogDesc rawLogDescription
-	if err := json.Unmarshal(data, &rawLogDesc); err != nil {
-		return fmt.Errorf("Failed to unmarshal log description, %s", err)
+// NewLog returns a initinalized Log struct
+func NewLog(uri, b64PK string) (*Log, error) {
+	var l Log
+	var err error
+	if strings.HasPrefix(uri, "/") {
```

Comment:
Google's client does [manual path concatenation](https://github.com/google/certificate-transparency/blob/master/go/client/logclient.go#L193) (instead of using say `path.Join`) and I'm pretty sure URI's containing "//" will break somewhere along the line.

This **should** be `HasSuffix` though... whops!